ot/solvers.py

```diff
@@ -412,7 +412,10 @@ def solve(
                 potentials = (log["u"], log["v"])
 
             elif reg_type.lower() in ["entropy", "kl"]:
-                if grad == "envelope":  # if envelope then detach the input
+                if grad in [
+                    "envelope",
```

Comment:
could you also add "detach" option here and in teh documengation? seems like a nice feature that can be practical